File: MyTestProject/Site.Models/Entities/Project.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.AspNetCore.Http;

namespace Site.Models.Entities
{
  public class Project : Entity<Guid>
  {
    public Guid OwnerId { get; set; } // creator id

    public string Name { get; set; }

    public string Description { get; set; }

    public string Content { get; set; }

    public string ImagePath { get; set; }


    //navigation property

    public List<UserProject> LinkedUsers { get; set; }// users linked to this project

    
  }
}

File: MyTestProject/Site.WEB/src/app/_models/project.ts
import { User } from './user';
import { UserProject } from './UserProject';

export class Project {

    id: string;
    content: string;
    description: string;
    name: string;
    linkedUsers: UserProject[];
    userId: string;
    user: User;

    imagePath: string;

    constructor(id?: string, content?: string, description?: string, name?: string, linkedUsers?: UserProject[], userId?: string, user?: User, imagePath?: string) {
        this.id = id;
        this.content = content;
        this.description = description;
        this.name = name;
        this.linkedUsers = linkedUsers;
        this.userId = userId;
        this.user = user;
        this.imagePath = imagePath;
    }

}
